### PR TITLE
[cpr] update to 1.8.1

### LINKED ISSUE #23981: [cpr] update to 1.8.1
Library name:
cpr

New version number:
1.8.1

Other information that may be useful

After the 1.7.1(Released the 09 Dec 2021)

2 version got published

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #23981
  Adds feature ssl

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/cpr/vcpkg.json [hunk 1]
```diff
@@ -16,5 +17,25 @@
...
+  ],
+  "default-features": [
+    "ssl"
+  ],
+  "features": {
+    "ssl": {
+      "description": "Enable SSL support",
+      "dependencies": [
+        {
+          "name": "curl",
```
Comment (dg0yt) [inline ports/cpr/vcpkg.json:29]: IMO there should be `"default-features": false` here, or feature `ssl` will always pull in the default features of `curl`.

File: ports/cpr/vcpkg.json [hunk 2]
```diff
@@ -16,5 +17,25 @@
...
+          "features": [
+            "ssl"
+          ]
+        },
+        {
+          "name": "openssl",
+          "platform": "linux"
```
Comment (dg0yt) [inline ports/cpr/vcpkg.json:36]: Does `cpr` call OpenSSL API, or is this a purely transitive usage requirement via `curl`?

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/cpr/vcpkg.json (hunk 2 above)
Comment (Thomas1664) [reply ports/cpr/vcpkg.json:36]: Yes, see its documentation

**Comment by JonLiu1993** [comment]
All feature are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**